### DISCUSSION TITLE
How to realize E2E transparent communication

### DISCUSSION THREAD

**Discussion by hktalent** [pion/webrtc#2186] [General] [answered]
Sorry, I still don't understand the existing example very well
I need an example like this now
1. First, a webrtc server is needed, which is only a P2P bridge for several clients
2. Then I need a demo of the client, this client allows to set a data CreateDataChannel to specify the key, all clients with the same key allow to send and receive data from each other
3. In fact, it realizes E2E transparent communication

**Comment by Sean-Der** [comment, marked as answer]
Hi @hktalent 

Have you seen https://github.com/pion/webrtc/tree/master/examples/insertable-streams 

Maybe start a repo based off of this and happy to answer any specific questions. WebRTC just provides the insertable streams API
